core/src/main/java/org/apache/iceberg/BaseEntriesTable.java

```diff
@@ -211,17 +211,32 @@ public <T> Boolean eq(BoundReference<T> ref, Literal<T> lit) {
         if (fileContent(ref)) {
           Literal<Integer> intLit = lit.to(Types.IntegerType.get());
           Integer fileContentId = intLit.value();
-          if (fileContentId == FileContent.DATA.id()) {
-            return manifestContentId == ManifestContent.DATA.id();
+          if (FileContent.DATA.id() == fileContentId) {
```

Comment:
As discussed we can refactor this into a common method:

```
contentMatch(int fileContentId, int manifestContentId) {
  if (fileContentId == FileContent.DATA.id()) {
      return manifestContentId == ManifestContent.DATA.id();
    } else if (fileContentId == FileContent.EQUALITY_DELETES.id()
        || fileContentId == FileContent.POSITION_DELETES.id()) {
      return manifestContentId == ManifestContent.DELETES.id();
    }
  }
  
  
Boolean eq(...)  {
   if (fileContent(ref)) {
     Literal<Integer> intLit = lit.to(Types.IntegerType.get());
     Integer fileContentId = intLit.value();
     if (!contentMatch(...)) return ROWS_CANNOT_MATCH;
   }
}

Boolean notEq(...) {
if (fileContent(ref)) {
   Literal<Integer> intLit = lit.to(Types.IntegerType.get());
   Integer fileContentId = intLit.value();
   if (contentMatch(...)) return ROWS_CANNOT_MATCH;
}
return ROWS_MIGHT_MATCH;
}

Reply:
thank you @szehon-ho , updated per your suggestion